stonesoup/sensor/radar/radar.py

```diff
@@ -0,0 +1,495 @@
+                         timestamp=ground_truth.timestamp)
+
+
+class RadarRotatingRangeBearing(RadarRangeBearing):
+    """A simple rotating radar, with set field-of-view (FOV) angle, range and\
+     rotations per minute (RPM), that generates measurements of targets, using\
+     a :class:`~.CartesianToBearingRange` model, relative to its\
+     position.
+
+    Note
+    ----
+    * The current implementation of this class assumes a 3D Cartesian plane.
+
+    """
+
+    dwell_center = Property(
+        State, doc="A state object, whose `state_vector`\
+        property describes the rotation angle of the center of the sensor's\
+        current FOV (i.e. the dwell center) relative to the positive x-axis\
+        of the sensor frame/orientation. The angle is positive if the rotation\
+        is in the counter-clockwise direction when viewed by an observer\
+        looking down the z-axis of the sensor frame, towards the origin.\
+        Angle units are in radians"
+    )
+    rpm = Property(
+        float,
+        doc="The number of antenna rotations per minute (RPM)")
+    max_range = Property(
+        float,
+        doc="The maximum detection range of the radar (in meters)")
+    fov_angle = Property(
+        float,
+        doc="The radar field of view (FOV) angle (in radians).")
+
+    def measure(self, ground_truth, noise=None, **kwargs):
+        """Generate a measurement for a given state
+
+        Parameters
+        ----------
+        ground_truth : :class:`~.State`
+            A ground-truth state
+
+        Returns
+        -------
+        :class:`~.Detection` or ``None``
+            A measurement generated from the given state, if the state falls\
+            in the sensor's field of view, or ``None``, otherwise. The\
+            timestamp of the measurement is set equal to that of the provided\
+            state.
+        """
+
+        # Read timestamp from ground truth
+        timestamp = ground_truth.timestamp
+
+        # Rotate the radar antenna and compute new heading
+        self.rotate(timestamp)
+        antenna_heading = self.orientation[2, 0] + \
+            self.dwell_center.state_vector[0, 0]
+
+        # Set rotation offset of underlying measurement model
+        rot_offset = \
+            StateVector(
+                [[self.orientation[0, 0]],
+                 [self.orientation[1, 0]],
+                 [antenna_heading]])
+
+        measurement_model = CartesianToBearingRange(
+            ndim_state=self.ndim_state,
+            mapping=self.mapping,
+            noise_covar=self.noise_covar,
+            translation_offset=self.position,
+            rotation_offset=rot_offset)
+
+        # Transform state to measurement space and generate
+        # random noise
+        measurement_vector = measurement_model.function(
+            ground_truth.state_vector, noise=0, **kwargs)
+
+        if noise is None:
+            measurement_noise = self.measurement_model.rvs()
```

Comment:
Need to remove `self.`

Reply:
This looks like something that was fixed in master after we branched off and didn't get merged to our branch as we have moved the file. I have reintroduced the fix and pushed.

Reply:
Also note that in master, line 149 is `if (noise is None):`. I have not re-introduced the redundant parentheses, as PyCharm would tell me off... This does make a change to a part of the code we shouldn't have touched, but seems ok to me. Let me know if you would prefer me to be strict about not touching the code (and hence re-introduce the parentheses).